asv_bench/benchmarks/utils/data_shapes.py

```diff
@@ -0,0 +1,186 @@
+    "TimeJoin": BINARY_SHAPES,
+    "TimeMerge": BINARY_SHAPES,
+    "TimeConcat": BINARY_SHAPES,
+    "TimeAppend": BINARY_SHAPES,
+    "TimeBinaryOp": BINARY_SHAPES,
+    "TimeSetItem": UNARY_SHAPES,
+    "TimeInsert": UNARY_SHAPES,
+    "TimeArithmetic": UNARY_SHAPES,
+    "TimeSortValues": UNARY_SHAPES,
+    "TimeDrop": UNARY_SHAPES,
+    "TimeHead": UNARY_SHAPES,
+    "TimeFillna": UNARY_SHAPES,
+    "TimeValueCountsFrame": UNARY_SHAPES,
+    "TimeValueCountsSeries": UNARY_SHAPES,
+    "TimeIndexing": UNARY_SHAPES,
+    "TimeMultiIndexing": UNARY_SHAPES,
+    "TimeResetIndex": UNARY_SHAPES,
+    "TimeAstype": UNARY_SHAPES,
+    "TimeDescribe": UNARY_SHAPES,
+    "TimeProperties": UNARY_SHAPES,
+    # IO benchmarks
+    "TimeReadCsvSkiprows": UNARY_SHAPES,
+    "TimeReadCsvTrueFalseValues": UNARY_SHAPES,
+    "TimeReadCsvNamesDtype": UNARY_SHAPES,
+    # Scalability benchmarks
+    "TimeFromPandas": UNARY_SHAPES,
+    "TimeToPandas": UNARY_SHAPES,
+    # OmniSci backend benchmarks
+    "omisci.TimeJoin": UNARY_SHAPES,
+    "omisci.TimeMerge": BINARY_SHAPES,
+    "omisci.TimeAppend": BINARY_SHAPES,
+    "omisci.TimeBinaryOpDataFrame": UNARY_SHAPES,
+    "omisci.TimeBinaryOpSeries": OMNISCI_SERIES_SHAPES,
+    "omisci.TimeArithmetic": UNARY_SHAPES,
+    "omisci.TimeSortValues": UNARY_SHAPES,
+    "omisci.TimeDrop": UNARY_SHAPES,
+    "omisci.TimeHead": UNARY_SHAPES,
+    "omisci.TimeFillna": UNARY_SHAPES,
+    "omisci.TimeValueCountsSeries": OMNISCI_SERIES_SHAPES,
+    "omisci.TimeIndexing": UNARY_SHAPES,
+    "omisci.TimeResetIndex": UNARY_SHAPES,
+    "omisci.TimeAstype": UNARY_SHAPES,
+    "omisci.TimeDescribe": UNARY_SHAPES,
+    "omisci.TimeProperties": UNARY_SHAPES,
+    "omisci.TimeGroupByDefaultAggregations": UNARY_SHAPES,
+    # IO benchmarks
+    "omnisci.TimeReadCsvNames": UNARY_SHAPES,
+}
+
+CONFIG_FROM_FILE = None
+
+
+def get_benchmark_shapes(bench_id: str):
+    """
+    Get custom benchmark shapes from a json file stored in MODIN_ASV_DATASIZE_CONFIG.
+
+    If `bench_id` benchmark is not found in the file, then the default value will
+    be used.
+
+    Parameters
+    ----------
+    bench_id : str
+        Unique benchmark identifier that is used to get shapes.
+
+    Returns
+    -------
+    list
+        Benchmark shapes.
+    """
+    global CONFIG_FROM_FILE
+    if not CONFIG_FROM_FILE:
+        try:
+            from modin.config import AsvDataSizeConfig
+
+            filename = AsvDataSizeConfig.get()
+        except ImportError:
+            filename = os.environ.get("MODIN_ASV_DATASIZE_CONFIG", None)
+        if filename:
+            # should be json
+            with open(filename) as _f:
+                CONFIG_FROM_FILE = json.load(_f)
+            if bench_id in CONFIG_FROM_FILE:
+                # convert strings to tuples;
+                # example: "omnisci.TimeReadCsvNames": ["(5555, 55)", "(3333, 33)"]
+                shapes = [eval(shape) for shape in CONFIG_FROM_FILE[bench_id]]
```

Comment:
Can we just use a proper json file which doesn't need an eval? It's completely valid json to have `[[5555, 555], [3333, 33]]`, why do you want to store them as opaque strings to be fed to `eval()`?

Reply:
Yes, we can.
`list` is now used everywhere instead of `tuple`.

I will also need to manually update the hash of the dashboards (to keep the old results displayed).